### PR TITLE
Add loaders cache

### PR DESCRIPTION
**Pull request by elronbandel**

### CODE REVIEW

File: src/unitxt/loaders.py [hunk 1]
```diff
@@ -132,10 +135,22 @@ def sef_default_data_classification(
...
+        pass
+
+    def _maybe_set_classification_policy(self):
         pass
 
+    def load_data(self):
+        iterables = self.__class__._loader_cache.get(str(self), None)
+        if iterables is None:
+            iterables = self.load_iterables()
+            self._loader_cache.max_size = settings.loader_cache_size
```
Comment (dafnapension) [inline src/unitxt/loaders.py:148]: maybe  self.__class__._cache.max_size    here too (stick the __class__ thing)?

File: src/unitxt/loaders.py [hunk 2]
```diff
@@ -132,10 +135,22 @@ def sef_default_data_classification(
...
 
     @abstractmethod
-    def load_data(self):
+    def load_iterables(self):
+        pass
+
+    def _maybe_set_classification_policy(self):
         pass
 
+    def load_data(self):
```
Comment (dafnapension) [inline src/unitxt/loaders.py:144]: def load_data(self)->MultiStream:

File: src/unitxt/loaders.py [hunk 3]
```diff
@@ -132,10 +135,22 @@ def sef_default_data_classification(
             self.data_classification_policy = default_data_classification_policy
 
     @abstractmethod
-    def load_data(self):
+    def load_iterables(self):
```
Comment (dafnapension) [inline src/unitxt/loaders.py:138]: def load_iterables(self)-> ?:

File: src/unitxt/loaders.py [hunk 4]
```diff
@@ -308,7 +312,7 @@ def load_data(self):
         if self.get_limit() is not None:
             return self.limited_load(dataset=dataset)
```
Comment (dafnapension) [inline src/unitxt/loaders.py:313]: here a MultiStream is returned

File: src/unitxt/utils.py [hunk 5]
```diff
@@ -18,6 +20,93 @@ def __call__(cls, *args, **kwargs):
...
+
+    @max_size.setter
+    def max_size(self, size):
+        with self._lock:
+            self._max_size = size
+            # Adjust the cache if the new size is smaller than the current number of items
+            while len(self._cache) > self._max_size:
+                self._cache.popitem(last=False)
+
+    def __setitem__(self, key, value):
```
Comment (dafnapension) [inline src/unitxt/utils.py:57]: is this needed in addition to `set` below?

File: src/unitxt/loaders.py (hunk 3 above)
Comment (elronbandel) [reply src/unitxt/loaders.py]: Done

**Review by dafnapension — APPROVED** [review]

File: src/unitxt/utils.py (hunk 5 above)
Comment (elronbandel) [reply src/unitxt/utils.py:57]: whats missing there?

**Pull request merged**